feat: 기업 검색 API 개발

dslov89 opened the pull request:
### Description

기업 검색 API 개발

<!--
  간단하게 PR의 목적을 설명하세요.
  이 PR이 해결하려는 문제나 추가하려는 기능에 대해 요약해주세요.
-->

### Related Issues

<!--
  관련된 이슈 번호를 참고하세요.
  예: Fixes #123, Closes #456
-->
- Resolves #29 

### Changes Made

<!--
  이 PR에서 변경된 사항을 설명하세요.
  코드, 문서, 설정 등 변경된 내용을 상세히 기술합니다.
-->

1. 각 폴더(service, domain, repository) 생성
2. 기업 검색 entitiy 작성
3. 기업 검색 api 개발

### Screenshots or Video

<!--
  변경된 사항이 UI에 영향을 미치는 경우, 변경 전후의 스크린샷이나 동영상을 첨부하세요.
-->

### Testing

<!--
  변경 사항을 테스트한 방법을 설명하세요.
  테스트한 환경 (OS, 브라우저, 장치 등)과 테스트 절차를 구체적으로 기술합니다.
-->

1. 스웨거로 제대로 작동하는 것을 확인

### Checklist

<!--
  PR 작성 시 다음 항목들을 확인하세요.
-->

- [x] 코드가 정상적으로 컴파일되는지 확인했습니다.
- [x] 모든 테스트가 성공적으로 통과했습니다.
- [x] 관련 문서를 업데이트했습니다.
- [x] PR이 관련 이슈를 정확히 참조하고 있습니다.
- [x] 코드 스타일 가이드라인을 준수했습니다.
- [x] 코드 리뷰어를 지정했습니다.

### Additional Notes

<!--
  리뷰어가 이해하는 데 도움이 될 추가적인 참고 사항이나 정보가 있다면 여기에 작성하세요.
-->

dslov89 committed:
feat: 기업 관련 폴더 구조 생성

기업 관련 폴더 - service, repository, model 생성

dslov89 committed:
ci: develop로 부터 병합

dslov89 committed:
feat: 기업 검색 API 개발

키워드 기준으로 select
jpa 내장 함수는 공백 제거가 되지 않아 직접 query 작성

Close #29

dslov89 committed:
refactor: Entity 패키지 및 파일 이름 변경

bakgom123 reviewed (changes requested):
인터페이스 부분 수정이 필요할 것 같습니다. 확인 부탁드려요.

dslov89 committed:
refactor: 서비스 인터페이스와 구현을 하나의 클래스로 통합

CompanyServiceImpl을 제거하고, 서비스 메서드를 CompanyService 클래스에 직접 통합
인터페이스 계층을 없애고 서비스 구조를 단순화
